back-end/scripts/api-injection.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+# Capture New URL as Argument
+new_url="$1"
+
+# Path to JavaScript File
+file="../../front-end/javascript/visitor-counter.js"
+
+# Replace Old URL in JavaScript File
+ sed -i "s#var apiUrl = \".*\"#var apiUrl = \"$new_url\"#" "$file"
+
+# Grabs URL From JavaScript File
+apiUrl=$(grep -oP 'apiUrl\s*=\s*"\K[^"]+' $file)
+
+# Compares URL's
+if [ "$apiUrl" == "$new_url" ]; then
+    echo "New API URL has been successfully injected."
+    exit 0
+else
+    echo "URL's are not the same" >&2
```

Comment:
Explain to me your understanding of what the ">&2" is doing in this script. I'm just curious if this is something ChatGPT gave you and you just ran with, or you fully understand what is going on?

Reply:
If I remember correctly, the ">&2" means you're redirecting the output to the standard error (stderr) output/stream, because if it didn't have the ">&2", it would just be a standard output (stdout). Why exactly it uses that specific syntax of ">&2" I have no idea.

Reply:
You nailed it honestly. Even with the 'Why exactly it uses that specific syntax of ">&2" I have no idea' -- it's not necessarily a bash thing, but nonetheless, welcome to bash :smile: